zerver/migrations/0174_set_initial_value_is_private_flag.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+import sys
+from django.db import migrations
+
+from django.db import migrations, transaction
+from django.db.backends.postgresql_psycopg2.schema import DatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.db.models import F
+
+def set_initial_value_of_is_private_flag(
+        apps: StateApps, schema_editor: DatabaseSchemaEditor) -> None:
+    UserMessage = apps.get_model("zerver", "UserMessage")
+    UserProfile = apps.get_model("zerver", "UserProfile")
+    user_profile_ids = UserProfile.objects.all().order_by("id").values_list("id", flat=True)
+    Subscription = apps.get_model("zerver", "Subscription")
+
+    i = 1
+    total = len(user_profile_ids)
+    print("\nStart setting initial value for is_private flag...")
+    sys.stdout.flush()
+    for user_id in user_profile_ids:
+        if (i % 50 == 0):
+            print("%s/%s" % (i, total))
+            sys.stdout.flush()
+        recipient_ids = Subscription.objects.filter(recipient__type__in=[1, 3], user_profile_id=user_id).\
+            values_list("recipient_id", flat=True)
+        with transaction.atomic():
+            private_objects = UserMessage.objects.filter(user_profile__id = user_id,
+                                                         message__recipient_id__in=recipient_ids)
+            private_objects.update(flags=F('flags').bitor(UserMessage.flags.is_private))
+        if (i % 50 == 0 or i == total):
+            percent = round((i / total) * 100, 2)
+            print("Processed %s/%s %s%%" % (i, total, percent))
+            sys.stdout.flush()
+        i += 1
```

Comment:
I've been thinking about this loop, and I think something built around this query would be a cleaner structure:

```
message_ids = Message.objects.filter(recipient__type__in=[1,3],                                            
                                             id__gte=range_start,                                          
                                             id__lte=range_end).values_list("id", flat=True).order_by("id")
UserMessage.objects.filter(message_id__in=message_ids).update(flags=F('flags').bitor(UserMessage.flags.is_private))

```

Reply:
Does this eliminate `for user_id in user_profile_ids:` loop and run the migrations in a batch of (e.g a loop of 10k messages) `k` messages?

Reply:
yes, exactly.